planetiler-core/src/main/java/com/onthegomap/planetiler/Planetiler.java

```diff
@@ -752,6 +764,50 @@ private void ensureInputFilesExist() {
     }
   }
 
+  // Helper for globMatchPath which will not recurse into zip files.
+  private List<Path> globMatchPath(Path basePath, String pattern) {
+    try {
+      return globMatchPath(basePath, pattern, zipPath -> List.of());
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  /**
+   * Returns a list of paths matching {@param pattern} within {@param basePath}, which may be either a ZIP archive or a
+   * directory.
+   * <p/>
+   * When {@param basePath} is a directory, the additional {@param onZipFile} function will be called whenever a ZIP
+   * archive matching {@param pattern} is found. This allows a different pattern to be used within the archive.
+   */
+  private List<Path> globMatchPath(Path basePath, String pattern, Function<Path, List<Path>> onZipFile)
```

Comment:
We might want to move this into FileUtils with a simple test or two just to verify it's doing what we expect without all of the mapping logic included.